File: app/src/main/java/com/yonder/addtolist/scenes/languageselection/domain/ContextWrapper.java
package com.yonder.addtolist.scenes.languageselection.domain;

import android.content.Context;
import android.content.res.Configuration;
import android.content.res.Resources;
import android.os.LocaleList;

import java.util.Locale;

/**
 * @author yusuf.onder
 * Created on 9.02.2022
 */


public class ContextWrapper extends android.content.ContextWrapper {

    public ContextWrapper(Context base) {
        super(base);
    }

    public static ContextWrapper wrap(Context context, Locale newLocale) {
        Resources res = context.getResources();
        Configuration configuration = res.getConfiguration();

        if (BuildUtils.INSTANCE.isAtLeast24Api()) {
            configuration.setLocale(newLocale);

            LocaleList localeList = new LocaleList(newLocale);
            LocaleList.setDefault(localeList);
            configuration.setLocales(localeList);

            context = context.createConfigurationContext(configuration);

        } else if (BuildUtils.INSTANCE.isAtLeast17Api()) {
            configuration.setLocale(newLocale);
            context = context.createConfigurationContext(configuration);

        } else {
            configuration.locale = newLocale;
            res.updateConfiguration(configuration, res.getDisplayMetrics());
        }

        return new ContextWrapper(context);
    }
}

File: app/src/main/java/com/yonder/addtolist/common/utils/ContextWrapper.kt
package com.yonder.addtolist.common.utils

import android.content.Context
import android.os.LocaleList
import com.yonder.addtolist.common.utils.device.BuildUtils.isAtLeast17Api
import com.yonder.addtolist.common.utils.device.BuildUtils.isAtLeast24Api
import java.util.*

/**
 * @author yusuf.onder
 * Created on 9.02.2022
 */
class ContextWrapper(base: Context?) : android.content.ContextWrapper(base) {
    companion object {
        fun wrap(context: Context, newLocale: Locale?): ContextWrapper {
            var context = context
            val res = context.resources
            val configuration = res.configuration
            when {
                isAtLeast24Api -> {
                    configuration.setLocale(newLocale)
                    val localeList = LocaleList(newLocale)
                    LocaleList.setDefault(localeList)
                    configuration.setLocales(localeList)
                    context = context.createConfigurationContext(configuration)
                }
                isAtLeast17Api -> {
                    configuration.setLocale(newLocale)
                    context = context.createConfigurationContext(configuration)
                }
                else -> {
                    configuration.locale = newLocale
                    res.updateConfiguration(configuration, res.displayMetrics)
                }
            }
            return ContextWrapper(context)
        }
    }
}